src/layout/IHyprLayout.cpp

```diff
@@ -313,9 +316,21 @@ void IHyprLayout::onMouseMove(const Vector2D& mousePos) {
     static auto PANIMATEMOUSE = CConfigValue<Hyprlang::INT>("misc:animate_mouse_windowdragging");
     static auto PANIMATE      = CConfigValue<Hyprlang::INT>("misc:animate_manual_resizes");
 
+    const auto  TIMERDELTA = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now() - TIMER).count();
+    const auto  MSDELTA    = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now() - MSTIMER).count();
+    const auto  MSMONITOR  = 1000.0 / g_pHyprRenderer->m_pMostHzMonitor->refreshRate;
+    static int  totalMs    = 0;
+
+    MSTIMER = std::chrono::high_resolution_clock::now();
+
+    if (m_iMouseMoveEventCount == 1)
+        totalMs = 0;
+
+    totalMs += MSDELTA < MSMONITOR ? MSDELTA : std::round(totalMs * 1.0 / m_iMouseMoveEventCount);
+    m_iMouseMoveEventCount += 1;
+
     if ((abs(TICKDELTA.x) < 1.f && abs(TICKDELTA.y) < 1.f) ||
-        (std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now() - TIMER).count() <
-         1000.0 / g_pHyprRenderer->m_pMostHzMonitor->refreshRate))
+        (TIMERDELTA < MSMONITOR && (MSMONITOR < 16.0 || std::clamp(MSMONITOR - TIMERDELTA, 0.0, MSMONITOR) > totalMs * 1.0 / m_iMouseMoveEventCount)))
```

Comment:
this is very confusing, some boolean variables or comments would help.

Reply:
smth like
```cpp
// if we face 60hz monitor, let's check wether remaing time before refresh is more than average input-events sending rate
```
ok?

Reply:
see, your code is so confusing you even confused yourself. I still recommend booleans.

`MSMONITOR > 16.0` if you wanna find 60Hz monitors. <16.0 targets everything _but_ 60hz panels.

Reply:
wait, we wan't orinial limiter to work, then it checks if monitor is **not** 60hz and returns, skipping average timing check
if it is 60hz (`MSMONITOR < 16.0` fails), perform one more check

Reply:
see, too confusing.